astropy/modeling/core.py

```diff
@@ -2462,8 +2460,16 @@ def render_model(model, arr=None, coords=None):
             assert coords[0].shape == arr.shape
         else:
             arr = np.zeros(coords[0].shape)
+    
+    if arr is not None:
+        arr = arr.copy()
+        # Check dimensions match model
+        assert arr.ndim == model.n_inputs
 
     if bbox is not None:
+        if wcs is not None:
+            bbox = np.array(wcs.wcs_world2pix(*[bbox[1],bbox[0],0]))[::-1]
+
```

Comment:
Yes, this is always painful to get right and needs quite a few tests.

Maybe as a first step, tonight you can explain to us what the semantics should be and, if you have time, show what currently happens for a few cases (e.g. at lat=+-90 or lon=0/360).